domains/pallets/executor-registry/src/lib.rs

```diff
@@ -693,11 +706,11 @@ mod pallet {
 
                         let stake_weight: T::StakeWeight = executor_config.stake.into();
 
-                        total_stake_weight = total_stake_weight.checked_add(&stake_weight).expect(
-                            "
-                                `total_stake_weight` as u128 won't overflow even with 100K executor and \
-                                each of them has 1_000_000_000 SSC at stake; qed",
-                        );
+                        total_stake_weight = total_stake_weight
+                            .checked_add(&stake_weight)
+                            .unwrap_or_else(|| {
+                                panic!("Domain bundle election can not work properly due to `total_stake_weight` overflow")
```

Comment:
Please return as Runtime Error so that we can recover easily just in case.

Reply:
Do you mean to move this piece of code out of `on_initialize`, and wrap it as a fallible function? You still have to handle the error inside `on_initialize` anyway, don't see how it can be recovered.

Reply:
ahh I see my bad, then can we just return MAX weight instead ?

Reply:
Maybe we can revert to previous state and continue. That would be another option

Reply:
Fair. I think we can guarantee non failure if we end up doing this check at the time of executor registration. If the total stake overflows, that particular txn will fail and in initialise we can gaurantee that through except

Reply:
That's doable but adds some complexity. We need to track a real-time total_stake_weight on each action that can lead to a change of total_stake_weight and ensure it doesn't overflow whenever it's about to be increased. Currently, there are 5 actions related: register, increase/decrease stake, pause/resume execution.

I'm okay if you all agree that's the direction _(although I feel such effort might be unnecessary considering the use case in the real world)_,  I'd love to make it an issue and address it in another PR later.

Reply:
If its gonna a big change, dont mind bringing those changes in a different pr with bit more thought around it.
Might as well revert to expect from panic for now :)

Reply:
Yes, the changes will probably not be trivial considering the expected tests.

expect is technically no different from panic, I can revert if you prefer keeping the code unchanged if it's not improved anyway.